engine/src/main/java/org/terasology/engine/recording/EventSystemReplayImpl.java

```diff
@@ -336,15 +336,16 @@ public void registerEventHandler(ComponentSystem handler) {
 
                 logger.debug("Found method: {}", method);
                 if (!Event.class.isAssignableFrom(types[0]) || !EntityRef.class.isAssignableFrom(types[1])) {
-                    logger.error("Invalid event handler method: {}", method.getName());
+                    logger.atError().addArgument(() -> method.getName()).log("Invalid event handler method: {}");
                     return;
                 }
 
                 requiredComponents.addAll(Arrays.asList(receiveEventAnnotation.components()));
                 List<Class<? extends Component>> componentParams = Lists.newArrayList();
                 for (int i = 2; i < types.length; ++i) {
                     if (!Component.class.isAssignableFrom(types[i])) {
-                        logger.error("Invalid event handler method: {} - {} is not a component class", method.getName(), types[i]);
+                        logger.atError().addArgument(() -> method.getName()).addArgument(types[i]).
+                                log("Invalid event handler method: {} - {} is not a component class");
```

Comment:
potentially called once for each parameter of every event handler method but only called in unexpected case and error-level log so unlikely that called "in vain"
```suggestion
                        logger.error("Invalid event handler method: {} - {} is not a component class", method.getName(), types[i]); //NOPMD
```